test/test.js

```diff
@@ -33,6 +34,33 @@ if (true) {
 
 });
 
+describe('node replacement', () => {
+    describe(`transform`, () => {
+        const nodeEnv = t.memberExpression(t.memberExpression(t.identifier('process'), t.identifier('env')), t.identifier('NODE_ENV'));
+        const nodeEnvCheck = t.binaryExpression('!==', nodeEnv, t.stringLiteral('production'));
+
```

Comment:
I think it's not a friendly api for users to understand babel node or babel-types . maybe we can make it easier to use. like 
```
plugins: [[plugin, {
   __DEV__: {
    type: 'node', // default to string
    replacement: 'process.env.NODE_ENV'
  }
}]]
```